packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -727,6 +727,28 @@ class HeroController extends NavigatorObserver {
     _maybeStartHeroTransition(route, previousRoute, HeroFlightDirection.pop, true);
   }
 
+  @override
+  void didStopUserGesture() {
+    if (navigator.userGestureInProgress)
+      return;
+
+    bool isStaleGesture(_HeroFlight flight) {
+      return flight.manifest.isUserGestureTransition
+          && flight.manifest.type == HeroFlightDirection.pop
+          && flight._proxyAnimation.value == 0
+          && flight._proxyAnimation.status == AnimationStatus.dismissed;
+    }
+
+    final List<_HeroFlight> invalidFlights = _flights.values
+      .where(isStaleGesture)
+      .toList(growable: false);
+
+    // Treat these invalidated flights as dismissed.
```

Comment:
Assuming that _handleAnimationUpdate() removes each invalid flight from _flights. Helpful to say as much here (it also explains why we applied toList() to `_flights.values.where...`